src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Metadata/IModelBindingMessageProvider.cs

```diff
@@ -30,5 +30,46 @@ public interface IModelBindingMessageProvider
         /// </summary>
         /// <value>Default <see cref="string"/> is "The value '{0}' is invalid.".</value>
         Func<string, string> ValueMustNotBeNullAccessor { get; }
+
+        /// <summary>
+        /// Fallback error message HTML and tag helpers display when a property is invalid but the
+        /// <see cref="ModelError"/>s have <c>null</c> <see cref="ModelError.ErrorMessage"/>s.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "The value '{0}' is invalid.".</value>
+        Func<string, string> ValueInvalid_UnknownErrorResource { get; }
+
+        /// <summary>
+        /// Replacement <see cref="ModelError.ErrorMessage"/> used in <see cref="ModelError"/> when
+        /// <see cref="ModelError.Exception"/> is of type <see cref="FormatException"/> and value is known i.e. when
+        /// the bound value could not be converted and the "replacer" has the attempted value.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "The value '{0}' is not valid for {1}.".</value>
+        Func<string, string, string> ValueInvalid_WithValueResource { get; }
+
+        /// <summary>
+        /// Replacement <see cref="ModelError.ErrorMessage"/> used in <see cref="ModelError"/> when
+        /// <see cref="ModelError.Exception"/> is of type <see cref="FormatException"/> and value is unknown i.e. when
+        /// the bound value could not be converted and the "replacer" does not have the attempted value.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "The supplied value is invalid for {0}.".</value>
+        Func<string, string> ValueInvalid_WithoutValueResource { get; }
+
+        /// <summary>
+        /// Fallback error message used when deserialization of request body fails due to no supported encodings
+        /// found on the input formatter.
+        /// </summary>
+        /// <value>
+        /// Default <see cref="string"/> is "No encoding found for input formatter '{0}'. There must be at
+        /// least one supported encoding registered in order for the formatter to read content.".
+        /// </value>
+        Func<string, string> NoEncodingFoundOnInputFormatter { get; }
```

Comment:
I have reverted this change now (also unlike other messages which are user facing or which are displayed to the user, the messages by `NoEncodingFoundOnInputFormatter` and `unsupportedcontenttype` are consumed by api clients.)